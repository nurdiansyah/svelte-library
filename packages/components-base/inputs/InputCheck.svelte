<script>
  /**
   *
   * @component InputCheck
   *
   */

  import { clsx, getId } from "../utils";
  import ButtonIcon from "./ButtonIcon.svelte";

  // state
  let { class: className } = $$props;

  /**
   * @type {string}
   */
  let classes;

  // props
  /**
   *
   * @type {boolean}
   */
  export let disabled = false;

  /**
   *
   * @type {boolean}
   */
  export let checked = false;

  /**
   *
   * @type {*}
   */
  export let buttonProps = {};

  /**
   *
   * @type {boolean}
   */
  export let focused = false;

  /**
   *
   * @type {?'small'}
   */
  export let size = undefined;

  /**
   * @type {string}
   */
  export let type = "checkbox";

  // handler

  // reactive
  $: classes = clsx(
    className,
    "input-check",
    size && `-size-${size}`,
    checked && "-checked",
    disabled && "-disabled",
    focused && "-focused"
  );
</script>

<span class={classes}>
  <ButtonIcon {...buttonProps} {disabled}>
    <input {...$$restProps} class="input" {type} {disabled} {checked} on:onchange on:blur on:click on:focus />
    <slot />
  </ButtonIcon>
  <slot name="track" />
</span>

<style src="./styles/input-check.scss" global>
</style>
