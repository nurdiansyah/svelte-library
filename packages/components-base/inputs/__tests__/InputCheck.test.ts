import { render as svelteRender } from "@testing-library/svelte";
import InputCheck from "../InputCheck.svelte";

describe("InputCheck", () => {
  const testId = "input-check";
  const props = { "data-testid": testId };
  const render = (_props = {}) => svelteRender(InputCheck, { ...props, ..._props });
  it("check: type checkbox", () => {
    const { container } = render();
    expect(container).toMatchSnapshot();
  });

  it("check: type radio", () => {
    const { container } = render({ type: "radio" });
    expect(container).toMatchSnapshot();
  });

  it("check: type switch", () => {
    const { container } = render({ type: "switch" });
    expect(container).toMatchSnapshot();
  });
});
