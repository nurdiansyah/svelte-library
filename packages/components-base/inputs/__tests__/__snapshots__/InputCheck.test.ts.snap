// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCheck check: type checkbox 1`] = `
<body>
  <div />
</body>
`;

exports[`InputCheck check: type radio 1`] = `
<body>
  <div />
</body>
`;

exports[`InputCheck check: type switch 1`] = `
<body>
  <div />
</body>
`;
